### PR TITLE
chore: payg adjustment estimator follow ups from review

### PR DESCRIPTION
**Pull request by simplesagar**
Follow-up to #5145, addressing the two cubic review comments there.

## What

1. **Uplift no longer suppresses the modelling-error warning.** The "cheaper at adjusted rates" delta message now requires a *discount* (`rateAdjustPct < 0`). Under an uplift, PAYG still undercutting the committed model is a stronger anomaly signal — an uplift only raises PAYG above list — so it falls through to the "check the platform fee against the baseline" warning.
2. **≤ −100% adjustments unified on the list-rate fallback.** The pure `paygLines` previously floored rates at zero (a free PAYG bill) while the estimator input already fell back to list rates for out-of-range entries. Both layers now agree: at or past −100% the swing is invalid and ignored, so nothing ever prices PAYG as free.

Also moves `paygDeltaMessage` / `formatSignedPct` into `contract-pricing.ts` so the branch logic is unit-testable (5 new tests; 31 total pass).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01L8edBG1tMBsaDSGPgi4pt9

---
## Summary by cubic
Fixes PAYG adjustment estimator edge cases: only valid discounts claim a cheaper-than-committed outcome, uplifts keep the modelling-error warning, and swings at or below −100% now fall back to list rates across pricing and messaging to prevent free or misattributed PAYG.

- **Bug Fixes**
  - A cheaper PAYG under an uplift now shows the modelling-error warning; the “cheaper at adjusted rates” message only appears for valid discounts (`rateAdjustPct < 0` and > −100%).
  - Swings ≤ −100% are treated as invalid and ignored; all helpers use list rates, so PAYG is never priced as free and messages don’t credit an unapplied adjustment.

- **Refactors**
  - Centralized rate-adjust validity in `isValidRateAdjust`; moved `paygDeltaMessage` and `formatSignedPct` into `contract-pricing.ts` with new unit tests.

<sup>Written for commit b8fbf51331596d4f4aa80a47bbe64fbb98a0befa. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/speakeasy-api/gram/pull/5146?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: client/dashboard/src/components/billing/contract-pricing.ts [hunk 1]
```diff
@@ -278,6 +280,31 @@ export function formatUSD(value: number): string {
...
+// line. PAYG undercutting the committed contract is a modelling error at
+// list rates — and a fortiori under an uplift, which only raises PAYG above
+// list — but it's the expected outcome of a big enough negotiated discount.
+// So only a discount claims the gap; everything else falls through to the
+// check-the-fee warning.
+export function paygDeltaMessage(delta: number, rateAdjustPct: number): string {
+  if (delta > 0) {
+    return `Pay-as-you-go costs ${formatUSD(delta)}/yr more than the committed contract at this volume — the number to point at in a "should we commit?" conversation.`;
+  }
+  if (delta < 0 && rateAdjustPct < 0) {
```
[the remark this answers is not in the document]
Comment (simplesagar) [reply client/dashboard/src/components/billing/contract-pricing.ts]: Fixed in b8fbf5133 — both functions now consult a shared isValidRateAdjust predicate, so the discount branch cannot claim a swing paygLines ignored.
